### ISSUE TITLE
Rule Change: Disallow specifying both `max` and `maximum` options

### ISSUE THREAD

**Issue by Pixel998** [eslint/eslint#20974]
### What rule do you want to change?

`complexity`, `max-depth`, `max-nested-callbacks`, `max-params`, and `max-statements`

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

Other

### Example code

```js
export default [
    {
        rules: {
            complexity: ["error", { max: 2, maximum: 5 }],
            "max-depth": ["error", { max: 2, maximum: 5 }],
            "max-nested-callbacks": ["error", { max: 2, maximum: 5 }],
            "max-params": ["error", { max: 2, maximum: 5 }],
            "max-statements": ["error", { max: 2, maximum: 5 }]
        }
    }
];
```

### What does the rule currently do for this code?

These rule schemas currently allow both `max` and `maximum` to be present in the same options object. In the rule implementations, the configured limit is resolved with `option.maximum || option.max`, so `maximum` effectively takes precedence when both values are provided.

### What will the rule do after it's changed?

The rule schemas should reject options objects that specify both `max` and `maximum`.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

Since rule schemas are part of ESLint's public API, this would be a breaking change.

**Comment by DMartens** [comment]
This was discussed for `max-params` and decided to be part of the next major version.
As such we should widen the existing issue to also handle the other rules.
